app/src/main/java/com/example/criminalintent/presentation/crime_detail/CrimeFragment.kt

```diff
@@ -171,18 +171,19 @@ class CrimeFragment : Fragment(), FragmentResultListener {
             RESULT_OK -> {
                 val contactURI: Uri? = result.data?.data
                 val queryFields = arrayOf(ContactsContract.Contacts.DISPLAY_NAME)
-                val cursor = requireActivity().contentResolver.query(
-                    contactURI!!, queryFields, null, null, null
-                )
+                val cursor = contactURI?.let {
```

Comment:
If this is null, nothing will be shown to the user and his action will not be completed.
You should inform the user that something went wrong when choosing the name

Reply:
Moreover, you are just moving the nullable value further down.
You should handle the null intent where you first encounter it.
If it is null, there is nothing more you can do